tests/simple/testdata/proto2.textproto

```diff
@@ -1,8 +1,436 @@
 name: "proto2"
 description: "Protocol buffer version 2 tests.  See notes for the available set of protos for tests."
+
 section {
-  name: "singular_read"
-  description: "Reading the singlular fields."
+  name: "literal_singular"
+  description: "Literals with singular fields set."
+
+  test {
+    name: "int64_nocontainer"
+    expr: "google.api.expr.test.v1.proto2.TestAllTypes{single_int64: 17}"
+    value {
+      object_value {
+        [type.googleapis.com/google.api.expr.test.v1.proto2.TestAllTypes]
+        { single_int64: 17 }
+      }
+    }
+  }
+
+  test {
+    name: "int32"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "TestAllTypes{single_int32: -34}"
+    value {
+      object_value {
+        [type.googleapis.com/google.api.expr.test.v1.proto2.TestAllTypes]
+        { single_int32: -34 }
+      }
+    }
+  }
+
+  test {
+    name: "int64"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "TestAllTypes{single_int64: 17}"
+    value {
+      object_value {
+        [type.googleapis.com/google.api.expr.test.v1.proto2.TestAllTypes]
+        { single_int64: 17 }
+      }
+    }
+  }
+
+  test {
+    name: "uint32"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "TestAllTypes{single_uint32: 1u}"
+    value {
+      object_value: {
+        [type.googleapis.com/google.api.expr.test.v1.proto2.TestAllTypes]
+        { single_uint32: 1 }
+      }
+    }
+  }
+
+  test {
+    name: "uint64"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "TestAllTypes{single_uint64: 9999u}"
+    value {
+      object_value {
+        [type.googleapis.com/google.api.expr.test.v1.proto2.TestAllTypes]
+        { single_uint64: 9999 }
+      }
+    }
+  }
+
+  test {
+    name: "sint32"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "TestAllTypes{single_sint32: -3}"
+    value {
+      object_value {
+        [type.googleapis.com/google.api.expr.test.v1.proto2.TestAllTypes]
+        { single_sint32: -3 }
+      }
+    }
+  }
+
+  test {
+    name: "sint64"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "TestAllTypes{single_sint64: 255}"
+    value {
+      object_value {
+        [type.googleapis.com/google.api.expr.test.v1.proto2.TestAllTypes]
+        { single_sint64: 255 }
+      }
+    }
+  }
+
+  test {
+    name: "fixed32"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "TestAllTypes{single_fixed32: 43u}"
+    value {
```

Comment:
Does the test pass or fail if value provided to the field is an `int`?

Reply:
Fails in type checking.  If type checking is disabled, we get an unsupported type conversion error at runtime. (It also triggers downstream errors in other tests due to different map field orders - need to fix the drivers to adjust for that!)